src/Middleware/IdleConnectionMiddleware.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\Middleware;
+
+use ArrayObject;
+use Doctrine\DBAL\Driver;
+use Doctrine\DBAL\Driver\Middleware;
+use Symfony\Bridge\Doctrine\Middleware\IdleConnection\Driver as IdleConnectionDriver;
+
+class IdleConnectionMiddleware implements Middleware, ConnectionNameAwareInterface
+{
+    private ArrayObject $connectionExpiries;
+    /** @var array<string, int> */
+    private array $ttlByConnection;
+    private string $connectionName;
+
+    /**
+     * @param ArrayObject<string, int> $connectionExpiries
+     * @param array<string, int>       $ttlByConnection
+     */
+    public function __construct(ArrayObject $connectionExpiries, array $ttlByConnection)
```

Comment:
```suggestion
    public function __construct(ArrayObject $expiringConnections, array $ttlByConnection)
```

Reply:
If I'm not mistaken, this is a list of expiries and not connections. There were discussions regarding the naming of variables above, and it was decided to use `connectionExpiries`